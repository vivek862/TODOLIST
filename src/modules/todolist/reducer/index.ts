const initialState = {
    tasks: {
      'task1': {id: 'task1', title: 'Task 1', content: 'This is a very good description'},
      'task2': {id: 'task2', title: 'Task 2', content: 'Mail to xxxxx on tuesday at 3.30'},
      'task3': {id: 'task3', title: 'Task 3', content: 'Call amber on friday'},
      'task4': {id: 'task4', title: 'Task 4', content: 'call madison on thursady'}
    },
    columns: {
      'column1': {
        id: 'column1',
        title: 'To Do',
        taskIds: ['task1', 'task2', 'task3', 'task4']
      }
    },
    columnIds: ['column1']
}; 
  export default function ToDoList(state = initialState, action) {
    switch (action.type) {
      case 'UPDATE_TASKS':
        return {...state, tasks: action.data};
      case 'ADD_COLUMN':
        const columns = {
          ...state.columns,
          [action.data.id]: action.data
        };
        const columnIds = state.columnIds;
        columnIds.push(action.data.id);   
        return {...state, columns: columns, columnIds: columnIds};
      case 'UPDATE_STATE':
        const newColumns = {
          ...state.columns,
          [action.data.id]: action.data
        }
        return {...state, columns: newColumns};   
      default:
        return state
    }
  }