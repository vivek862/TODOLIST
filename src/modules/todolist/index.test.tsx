import React from 'react';
import { shallow } from 'enzyme';
import { App } from './index';
import toJson from 'enzyme-to-json'
import { initialState } from './reducer/index'

describe("Testing App Component", () => {
    it("App Component Should Render", () => {
        const props = {
            toDoList: initialState,
            toDoActions:{
                addColumnData: jest.fn(),
                updateColumnData: jest.fn()
            }
          }
          const wrapper = shallow(<App {...props} />)
          expect(toJson(wrapper)).toMatchSnapshot();
        });
    })