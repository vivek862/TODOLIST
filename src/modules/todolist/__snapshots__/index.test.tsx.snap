// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component App Component Should Render 1`] = `
<div
  className="topLevelEmenet"
>
  <div
    className="title"
  >
    To Do App
  </div>
  <div
    className="columns"
  >
    <DragDropContext
      onDragEnd={[Function]}
    >
      <Column
        column={
          Object {
            "id": "column1",
            "taskIds": Array [
              "task1",
              "task2",
              "task3",
              "task4",
            ],
            "title": "To Do",
          }
        }
        key="column1"
        tasks={
          Array [
            Object {
              "content": "This is a very good description",
              "id": "task1",
              "title": "Task 1",
            },
            Object {
              "content": "Mail to xxxxx on tuesday at 3.30",
              "id": "task2",
              "title": "Task 2",
            },
            Object {
              "content": "Call amber on friday",
              "id": "task3",
              "title": "Task 3",
            },
            Object {
              "content": "call madison on thursady",
              "id": "task4",
              "title": "Task 4",
            },
          ]
        }
      />
      <AddDetail
        text="+ Add Column"
        updateData={[Function]}
      />
    </DragDropContext>
  </div>
</div>
`;
